File: March 5, 2026.js
function smallestGap(str) {

  let lastSeen = {};
  let result = {};
  let minDistance = 100000000000;

  for (let i = 0; i < str.length; i++) {
    let char = str[i];

    if (lastSeen[char] !== undefined) {
      let distance = i - lastSeen[char];

      if (distance < minDistance) {
        minDistance = distance;
        result = {
          minChar: char,
          positions: [lastSeen[char], i],
          distance: distance
        };
      }
    }

    lastSeen[char] = i;
  }

  return str.slice(result.positions[0] + 1, result.positions[1]);
}

File: March 5, 2026.py
def smallest_gap(s):

    lastSeen = {}
    result = {}
    minDistance = 10000000000

    for i in range(len(s)):
        char = s[i]

        if (char in lastSeen):
            distance = i - lastSeen[char]

            if (distance < minDistance):
                minDistance = distance
                result = {
                    "minChar": char,
                    "positions": [lastSeen[char], i],
                    "distance": distance
                }
        lastSeen[char] = i

    return s[result["positions"][0] + 1 : result["positions"][1]]